hypervisor/src/hypervisor.rs

```diff
@@ -118,3 +118,13 @@ pub trait Hypervisor: Send + Sync {
     #[cfg(feature = "tdx")]
     fn tdx_capabilities(&self) -> Result<TdxCapabilities>;
 }
+///
+/// Generic MemoryRegion struct
+///
+pub struct MemoryRegion {
+    pub slot: u32,
```

Comment:
`Slot` is a KVM specific detail. You don't need to expose it.

I can see how removing `slot` can be cumbersome for other places. For the sake of simplicity you can keep using `slot` for now.